From f48a16d42eee30730f03a02626fbe96799b57a6b Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Wed, 12 Apr 2017 15:29:50 -0700
Subject: [PATCH 2/2] ARM: dts: imx6qdl-apalis: configure LVDS VGA and dual
 mdoe

Configure dual mode, which allows displaying one image stream
duplicated to both channels (LVDS0 = LVDS1) with standard VGA
resolution.

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 arch/arm/boot/dts/imx6qdl-apalis.dtsi | 24 +++++++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-apalis.dtsi b/arch/arm/boot/dts/imx6qdl-apalis.dtsi
index 4708f892c234..7838f5b62205 100644
--- a/arch/arm/boot/dts/imx6qdl-apalis.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-apalis.dtsi
@@ -1009,7 +1009,7 @@
 &ldb {
 	status = "okay";
 //	split-mode;
-//	dual-mode;
+	dual-mode;
 
 	lvds-channel@0 {
 		reg = <0>;
@@ -1020,7 +1020,25 @@
 		status = "okay";
 
 		display-timings {
-			native-mode = <&timing_xga>;
+			native-mode = <&timing_vga>;
+
+			timing_vga: 640x480 {
+				clock-frequency = <25175000>;
+				hactive = <640>;
+				vactive = <480>;
+				hback-porch = <40>;
+				hfront-porch = <24>;
+				vback-porch = <32>;
+				vfront-porch = <11>;
+				hsync-len = <96>;
+				vsync-len = <2>;
+
+				de-active = <1>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				pixelclk-active = <0>;
+			};
+
 			/* LDB-AM-800600LTNQW-A0H */
 			timing_svga: 800x600 {
 				clock-frequency = <55000000>;
@@ -1086,7 +1104,7 @@
 		reg = <1>;
 		fsl,data-mapping = "spwg";
 		fsl,data-width = <18>;
-		crtc = "ipu1-di0";
+		crtc = "ipu2-di1";
 		status = "okay";
 
 		display-timings {
-- 
2.12.1

